src/pyhf/readxml.py

```diff
@@ -22,12 +22,14 @@ def extract_error(h):
     bin uncertainties are then Poisson, and so the `sqrt(entries)`.
 
     Args:
-        h (uproot.rootio.TH1 object): The histogram
+        hist (uproot.rootio.TH1 object): The histogram
```

Comment:
@jpivarski If I just add a

```python
print(hist, type(hist))
```

in the function below and use it on a test case I get

```python
<TH1F (version 2) at 0x7f8a9ee37b90> <class '<dynamic>.Model_TH1F_v2'>
```

Though if I search around in `uproot4` repo it seems that these are streamers that inherit from `Model`? I actually never looked into how `uproot3` did things either, but if we wanted to give a class description here then would 

```
hist(`uproot4.Model_TH1F_v2`)
```
?

Reply:
Uproots 3 and 4 both dynamically create classes from streamers. In Uproot 4, it's more formal in that they all inherit from Model. If you want to verify that something is a histogram, you can check to see if it inherits from `uproot4.behaviors.TH1.Model_TH1` (approximate: please check!) because the behavioral mix-ins (what gives it the `.to_numpy()` method, etc.) are non-dynamic classes that you can count on existing.

Reply:
Or if the class doesn't have any behavioral overloads, the correct way to do it would be a string comparison on its `classname` (which is a C++ class name, with templates). I don't know if there's a method that checks inheritance through the C++ class names, but there should be..

Reply:
(in Uproot 3, the existence of `uproot.rootio.TH1` was an unintentional byproduct of the way the dynamic class was generated.)